frontend/src/composables/useShootContext.js

```diff
@@ -222,21 +223,44 @@ export function createShootContextComposable (options = {}) {
     },
     set (value) {
       set(manifest.value, ['spec', 'secretBindingName'], value)
+      unset(manifest.value, ['spec', 'credentialsBindingName'])
       resetPurpose()
     },
   })
 
-  const infrastructureSecretBinding = computed({
+  const credentialsBindingName = computed({
     get () {
-      return find(infrastructureSecretBindings.value, ['metadata.name', secretBindingName.value])
+      return get(manifest.value, ['spec', 'credentialsBindingName'])
     },
     set (value) {
-      secretBindingName.value = get(value, ['metadata', 'name'])
+      set(manifest.value, ['spec', 'credentialsBindingName'], value)
+      unset(manifest.value, ['spec', 'secretBindingName'])
+      resetPurpose()
+    },
+  })
+
+  const infrastructureBinding = computed({
+    get () {
+      if (secretBindingName.value) {
+        return find(infrastructureBindings.value, { kind: 'SecretBinding', metadata: { name: secretBindingName.value } })
+      }
+      if (credentialsBindingName.value) {
+        return find(infrastructureBindings.value, { kind: 'CredentialsBinding', metadata: { name: credentialsBindingName.value } })
+      }
+      return undefined
+    },
+    set (value) {
+      if (value?.kind === 'CredentialsBinding') {
+        credentialsBindingName.value = get(value, ['metadata', 'name'])
+      }
+      if (value?.kind === 'SecretBinding') {
+        secretBindingName.value = get(value, ['metadata', 'name'])
+      }
```

Comment:
do we have to reset the other respective value?

Reply:
It does get resetted in the credentialsBindingName computed property setter